### PR TITLE
Use keyword arguments for at-cuda.

### LINKED ISSUE #94: cuda macro requires a literal tuple
Extremely minor, but I just figured I'd note all these small usability issues -- feel free to ignore if it's not what you're going for right now.

This has become a very common pattern for me:

```julia
blk, thr = cudims(length(C))
@cuda (blk, thr) ...
```

which is a bit boilerplate-y. I think the overhead of keyword args, noted in the code, can fairly easily be avoided by splatting a tuple -- or perhaps it will just get fixed with named tuples in future.

### PR DESCRIPTION
**Pull request by maleadt**
Fixes #94
Requires https://github.com/JuliaGPU/CUDAdrv.jl/pull/88

### CODE REVIEW

**Comment by maleadt** [comment]
@vchuravy df91294b2e5aa25a7f3ccba9dd6029af1526d11a replaces #149

File: src/execution.jl [hunk 1]
```diff
@@ -41,45 +41,56 @@ isghosttype(dt) = !dt.mutable && sizeof(dt) == 0
...
-@generated function _cuda(dims::Tuple{CuDim, CuDim}, shmem, stream,
-                          func::Core.Function, argspec...)
+@generated function _cuda(func::Core.Function, argspec...; kwargs...)
     arg_exprs = [:( argspec[$i] ) for i in 1:length(argspec)]
     arg_types = argspec
 
+    # split kwargs, only some are dealt with by the compiler
+    nt = kwargs.parameters[4]
+    kws = Base._nt_names(nt)::NTuple{N,Symbol} where {N}
+    compile_kws = Symbol[kws...] ∩ (:minthreads,:maxthreads)    # JuliaLang/julia#25801
```
Comment (vchuravy) [inline src/execution.jl]: There is also `minctasm` which is the second argument to `__launch_bounds__`. So far as I can tell nobody set `reqntid*`, at least Clang just leaves it be.

Comment (maleadt) [reply src/execution.jl]: Added. I don't think I'm going to support something like `__launch_bounds__` though, `minthreads` and `blocks_per_sm` etc look more user-friendly to me.

File: src/execution.jl [hunk 2]
```diff
@@ -109,6 +109,7 @@ const compilecache = Dict{UInt, CuFunction}()
         end
 
         # compile the function
+        # TODO: include maxthreads/minthreads in the hash?
```
[the remark this answers is not in the document]
Comment (maleadt) [reply src/execution.jl]: Done.

File: src/execution.jl [hunk 3]
```diff
@@ -84,7 +84,7 @@ const compilecache = Dict{UInt, CuFunction}()
     # split kwargs, only some are dealt with by the compiler
     nt = kwargs.parameters[4]
     kws = Base._nt_names(nt)::NTuple{N,Symbol} where {N}
-    compile_kws = kws ∩ (:maxthreads,)
+    compile_kws = Symbol[kws...] ∩ (:minthreads,:maxthreads)    # JuliaLang/julia#25801
```
[the remark this answers is not in the document]
Comment (maleadt) [reply src/execution.jl]: Added.

**Comment by maleadt** [comment]
So this'll mean the end of 0.6 compatibility, but 0.7 is close enough at this point. I might keep a `release-0.6` branch active if people care.

**Pull request merged**